### PR TITLE
Removed the returning of void return types as per job contract

### PR DESCRIPTION
**Pull request by johnathanmdell**
I've removed the return statements for methods within the queue ecosystem that should be void according to the job contract.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
Please send to master. :)

File: src/Illuminate/Queue/Worker.php [hunk 1]
```diff
@@ -539,9 +540,9 @@ protected function markJobAsFailedIfWillExceedMaxExceptions($connectionName, $jo
      * @param  \Throwable  $e
      * @return void
      */
-    protected function failJob($job, Throwable $e)
+    protected function failJob(Job $job, Throwable $e)
```
Comment (GrahamCampbell) [inline src/Illuminate/Queue/Worker.php:543]: BC break

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

**Comment by johnathanmdell** [comment]
Looks like the tests are failing due to a return being called on the mock.

Fixing the test.

File: src/Illuminate/Queue/Worker.php (hunk 1 above)
Comment (johnathanmdell) [reply src/Illuminate/Queue/Worker.php:543]: Noted

File: src/Illuminate/Queue/CallQueuedHandler.php [hunk 2]
```diff
@@ -230,10 +230,10 @@ protected function handleModelNotFound(Job $job, $e)
         }
 
         if ($shouldDelete) {
-            return $job->delete();
+            $job->delete();
```
Comment (rodrigopedra) [inline src/Illuminate/Queue/CallQueuedHandler.php:233]: Hi @johnathanmdell 

I don't know if you are going to send it to master as suggested by @GrahamCampbell , but removing the return here also changes the code execution.

Before this method would not continue execution if `$shouldDelete` evaluated to true, skipping the call for `$job->fail($e)`.

**Pull request closed without merging**